__tests__/index.test.js

```diff
@@ -289,4 +289,62 @@ describe('DELETE /api/comments/:comment_id', () => {
             expect(response.body.msg).toBe('Bad request')
         })
     });
+});
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+describe('GET /api/articles (topic query)', () => {
+    test('GET: 200 filters articles by topic specified in query', () => {
+        return request(app)
+        .get('/api/articles?topic=cats')
+        .expect(200)
+        .then((response) => {
+            expect(response.body.articles.length).toBe(1)
+            response.body.articles.forEach((article) => {
+                expect(article.topic).toBe('cats');
+                expect(article).toMatchObject({
+                    author: expect.any(String),
+                    title: expect.any(String),
+                    article_id: expect.any(Number),
+                    topic: expect.any(String),
+                    created_at: expect.any(String),
+                    votes: expect.any(Number),
+                    article_img_url: expect.any(String),
+                    comment_count: expect.any(String)
+                })
+                expect(article).not.toMatchObject({
+                    body: expect.any(String)
+                })
+            })
+        })
+    });
+    test('GET:404 sends an appropriate status and error message when given a valid but non-existent topic ', () => {
+        return request(app)
+        .get('/api/articles?topic=dogs')
+        .expect(404)
+        .then((response) => {
+            expect(response.body.msg).toBe('Topic does not exist')
+        })
+    });
```

Comment:
Really good testing so far, but there is a missing test that should be added here. 

What should happen when the user queries for a valid topic, but one which has no associated articles?